src/main/java/bio/terra/cda/app/controller/QueryApiController.java

```diff
@@ -72,4 +74,25 @@ public QueryApiController(
 
     return sendQuery(querySql, offset, limit, dryRun);
   }
+
+  @Override
+  public ResponseEntity<QueryResponseData> uniqueValues(
+      String version, @Valid String body, @Valid Integer limit) {
+
+    String bqTable = "gdc-bq-sample";
+    Map<String, String> tableParts = CDAUtils.parseTableName(body);
+    logger.info("TABLE_PARTS " + tableParts.toString());
+    // SELECT * FROM UNNEST (SELECT DISTINCT p.column FROM qualifiedTABLE as p);
+    String querySql =
+        "SELECT DISTINCT p."
```

Comment:
I don't think you need the `p.` ... `AS p` for this query, but I would try it out in the BQ console or bq command line tool to see what works.